apps/admin/backend/src/cvr_files.ts

```diff
@@ -622,32 +620,44 @@ export async function addCastVoteRecordReport({
         ).BatchLabel,
       });
 
-      // Add referenced images and layouts to the cast vote record
+      // Add the cast vote record to the store
+      const cvrData = JSON.stringify(legacyCastVoteRecord);
+      const addCastVoteRecordResult = store.addCastVoteRecordFileEntry(
+        electionId,
+        fileId,
+        cvr.UniqueId as BallotId,
+        cvrData
+      );
+      if (addCastVoteRecordResult.isErr()) {
+        return err({
+          type: 'ballot-id-already-exists-with-different-data',
+          index: castVoteRecordIndex,
+        });
+      }
```

Comment:
Well, it's my existing code that has been moved so it's a fair point. I wasn't worried about it since it's the only type of error that method can return, but I've added a `throwIllegalValue`check in e8d537c77 to keep it safer.